### PR TITLE
Added code snippets for Key Vault Administration clients and builders.

### PR DESCRIPTION
**Pull request by vcolin7**
Partially addresses: #7154.

Also updated the README and added embedded samples to it.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-administration/README.md [hunk 1]
```diff
@@ -252,87 +291,118 @@ The following sections provide several code snippets covering some of the most c
...
+    .subscribe(roleDefinition ->
+        System.out.printf("Retrieved role definition with name '%s' and role name '%s'.%n",
+            roleDefinition.getName(), roleDefinition.getRoleName()));
+```
+
+### Delete a role definition asynchronously
+Delete a role definition. To do this, the 'name' property from an existing role definition is required.
 
-accessControlAsyncClient.listRoleAssignments(roleAssignmentScope))
+<!-- embedme src/samples/java/ReadmeSamples.java#172-175 -->
```
Comment (srnagar) [inline sdk/keyvault/azure-security-keyvault-administration/README.md:330]: The path to `ReadmeSamples.java` seems to be incorrect (package name should be included in the path).

**Review by srnagar — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-administration/README.md [hunk 2]
```diff
@@ -494,6 +595,7 @@ This project has adopted the [Microsoft Open Source Code of Conduct][microsoft_c
 [maven]: https://maven.apache.org/
 [azure_subscription]: https://azure.microsoft.com/
 [azure_keyvault]: https://docs.microsoft.com/azure/key-vault/quick-create-portal
+[azure_keyvault_mhsm]: https://docs.microsoft.com/en-us/azure/key-vault/managed-hsm/quick-create-cli
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-administration/README.md:598]:
```suggestion
[azure_keyvault_mhsm]: https://docs.microsoft.com/azure/key-vault/managed-hsm/quick-create-cli
```

**Pull request merged**